aten/src/ATen/native/quantized/cpu/fake_quantize_per_tensor_affine.cpp

```diff
@@ -21,21 +22,19 @@ namespace {
 Notes:
   - quant_delay might be set to non-zero to help weights stabilize in the
     beginning of the training.
-  - quantization range [0, 2^bits - 1]
+  - quantization range [quant_min, quant_max]
 */
-
-class FakeQuantizePerTensorAffineOp_forward : public c10::OperatorKernel {
- public:
-  at::Tensor operator()(
-      at::Tensor X,
+Tensor fake_quantize_per_tensor_affine_cpu(
+      const Tensor& self,
       double scale,
       int64_t zero_point,
-      int64_t quant_min = 0,
-      int64_t quant_max = 255,
-      int64_t quant_delay = 0,
-      int64_t iter = 0
+      int64_t quant_min,
+      int64_t quant_max,
+      int64_t quant_delay,
+      int64_t iter
     ) {
     // Sanity checks.
+    TORCH_CHECK(self.scalar_type() == ScalarType::Float);
     if (quant_min > quant_max) {
```

Comment:
btw, why all those aren't just TORCH_CHECK ?

Reply:
oh, it's copied from previous implementation, will change.